pkg/types/validation/installconfig.go

```diff
@@ -229,3 +233,39 @@ func validatePlatform(platform *types.Platform, fldPath *field.Path, openStackVa
 	}
 	return allErrs
 }
+
+func validateProxy(p *types.Proxy, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if p.HTTPProxy == "" && p.HTTPSProxy == "" {
+		allErrs = append(allErrs, field.Required(fldPath, "Must include httpProxy or httpsProxy."))
+	}
+	if p.HTTPProxy != "" {
+		if err := validate.URI(p.HTTPProxy); err != nil {
```

Comment:
maybe extend the URI to check the scheme of the URI...?

Reply:
I created `URIWithProtocol` to do this. Let me know if that is not what you had in mind or suggestions.